packages/common/core-utils/src/assert.ts

```diff
@@ -9,29 +9,35 @@
  * @param condition - The condition that should be true, if the condition is false an error will be thrown.
  * Only use this API when `false` indicates a logic error in the problem and thus a bug that should be fixed.
  * @param message - The message to include in the error when the condition does not hold.
- * A number should not be specified manually: use a string.
+ * A number should not be specified manually: use a string literal instead.
  * Before a release, policy-check should be run, which will convert any asserts still using strings to
  * use numbered error codes instead.
+ * @param debugMessageBuilder - An optional function that can be used to build a debug message to include in the error in development builds.
+ * Only executed if `condition` is false. `debugMessageBuilder` is not executed in production builds, see `skipInProduction` for details.
  * @remarks
  * Use this instead of the node 'assert' package, which requires polyfills and has a big impact on bundle sizes.
  *
  * Assertions using this API will be included in all configurations: there is no option to disable or optimize them out.
  * Thus this API is suitable for detecting conditions that should terminate the application and produce a useful diagnostic message.
  * It can be used to ensure bad states are detected early and to avoid data corruption or harder to debug errors.
  *
- * In cases where the assert is very unlikely to have an impact on production code but is still useful as documentation and for debugging, consider using `debugAssert` instead
+ * In cases where the assert is very unlikely to have an impact on production code but is still useful as documentation and for debugging, consider using {@link debugAssert} instead
  * to optimize bundle size.
  *
  * This API is not intended for use outside of the Fluid Framework client codebase: it will most likely be made internal in the future.
  * @privateRemarks
  * This should be deprecated (as a non internal API) then moved to purely internal.
```

Comment:
Do we have an idea of if / how prevalently this is still used by our partners? I would love to finally make this internal...

Reply:
No idea: I haven't checked. The old one duplicated assert in common/lib/common-utils/src/assert.ts is internal, so I'm not sure why it still exists either: I assumed most customers would still be using that, but I guess not.

Reply:
Did a quick search in the Loop codebase - they have 76 modules that import this. @anthony-murphy do you happen to know if they have an alternative `assert` library that we should be migrating them to? I would love to make this internal...